### DISCUSSION TITLE
PSAT CLI Report | Accessing + saving + sharing

### DISCUSSION THREAD

**Discussion by helenafranca** [GoogleChromeLabs/ps-analysis-tool#647] [Ideas]
1. Accessing
Problem: Currently, reports are lost when I close the CMD prompt.
Solution: Automatically save all reports locally for easy future reference.

2. Saving
Problem: Saved reports only include the cookies tab for individual pages.
Solution: Include the cookies tab for the aggregated report in excel format + enable saving the insights report as a PDF.

3. Sharing
Problem: I'm unable to share the report with others.
Solution: Add a "Share" button to generate a shareable report link.

**Comment by fellyph** [comment]
**Hi Helena, Thank you for your valuable feedback!**

We appreciate you taking the time to outline these issues and suggest thoughtful solutions. Here’s how we plan to address them:

> 1. Accessing  
> Problem: Reports are lost when the CMD prompt is closed.

Automatically saving all the reports will lead to much-wasted space, and reports might be difficult to track later. Instead we suggest you save the reports you want by using the download report button, as shown in the screenshot:

<img width="899" alt="Screenshot 2024-05-02 at 17 11 13" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/330792/0d267969-7d98-404a-9dbd-7a4cfd93eaa4">

> 2. Saving  
> Problem: Saved reports only include the cookies tab for individual pages.

We're actively working on extending the functionality to include the cookies tab for the aggregated report, as well as enabling the saving of the insights report as a PDF. You can stay updated on its progress on GitHub issue #627.


> 3. Sharing  
> Problem: I'm unable to share the report with others.

Instead of adding a shareable link feature, our current roadmap already includes a feature that will allow users first to save the reports as mentioned above and later upload those reports to render the results of previous CLI runs. You can track its progress on GitHub issue [#616](https://github.com/GoogleChromeLabs/ps-analysis-tool/issues/616)
